api-get-object.go

```diff
@@ -289,6 +292,21 @@ func (o *Object) doGetRequest(request getRequest) (getResponse, error) {
 		return response, response.Error
 	}
 
+	// Check if ETag has changed since last request.
+	switch {
+	// If response doesn't carry a valid ETag, ignore.
+	case response.objectInfo.ETag == "":
+
+	// Set Object's etag on first request.
+	case o.etag == "":
+		o.etag = response.objectInfo.ETag
+
+	// Check if etag from response matches Object's etag.
+	case o.etag != response.objectInfo.ETag:
+		return response, syscall.EBADF
```

Comment:
@krisis @harshavardhana mentioned that he discussed this with you. The reason `ErrInvalidObject("ETag has changed")` is better than `syscall.EBADF` is if application logs the error it will know exactly why error was returned (instead of a generic bad FD)